python/pyspark/sql/tests.py

```diff
@@ -3344,6 +3342,22 @@ def test_vectorized_udf_wrong_return_type(self):
                     'Invalid.*type.*string'):
                 df.select(f(col('x'))).collect()
 
+    def test_vectorized_udf_decorator(self):
+        from pyspark.sql.functions import pandas_udf, col
+        df = self.spark.range(10)
+
+        @pandas_udf(returnType=LongType())
+        def identity(x):
+            return x
+        res = df.select(identity(col('id')))
+        self.assertEquals(df.collect(), res.collect())
+
+    def test_vectorized_udf_empty_partition(self):
+        from pyspark.sql.functions import pandas_udf, col
+        df = self.spark.createDataFrame(self.sc.parallelize([Row(id=1)], 2))
```

Comment:
Maybe I miss something, but what this test is intended to test?

Reply:
Oh. I see. One partition is empty and it is related to the added stuff in `ArrowEvalPythonExec`.

Reply:
Yeah, an empty partition leads to an empty iterator, so this is to make sure it can handle that.